src/packagedcode/opam.py

```diff
@@ -0,0 +1,309 @@
+        name=name or None,
+        vcs_url=vcs_url or None,
+        homepage_url=homepage_url,
+        description=description,
+        parties=parties,
+        dependencies=package_dependencies
+    )
+
+    return package
+
+"""
+Example:- 
+
+Sample opam file(sample3.opam):
+opam-version: "2.0"
+version: "4.11.0+trunk"
+synopsis: "OCaml development version"
+depends: [
+  "ocaml" {= "4.11.0" & post}
+  "base-unix" {post}
+]
+conflict-class: "ocaml-core-compiler"
+flags: compiler
+setenv: CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"
+build: [
+  ["./configure" "--prefix=%{prefix}%"]
+  [make "-j%{jobs}%"]
+]
+install: [make "install"]
+maintainer: "caml-list@inria.fr"
+homepage: "https://github.com/ocaml/ocaml/"
+bug-reports: "https://github.com/ocaml/ocaml/issues"
+authors: [
+  "Xavier Leroy"
+  "Damien Doligez"
+  "Alain Frisch"
+  "Jacques Garrigue"
+] 
+
+>>> p = parse_opam('sample3.opam')
+>>> for k, v in p.items():
+>>>     print(k, v)
+
+Output:
+opam-version 2.0
+version 4.11.0+trunk
+synopsis OCaml development version
+depends [Opam(name='ocaml', version='= 4.11.0 & post'), Opam(name='base-unix', version='post')]
+conflict-class ocaml-core-compiler
+flags compiler
+setenv CAML_LD_LIBRARY_PATH = %{lib}%/stublibs
+build 
+install make install
+maintainer ['caml-list@inria.fr']
+homepage https://github.com/ocaml/ocaml/
+bug-reports https://github.com/ocaml/ocaml/issues
+authors ['Xavier Leroy', 'Damien Doligez', 'Alain Frisch', 'Jacques Garrigue']
+"""
+
+@attr.s()
+class Opam(object):
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+
+    @property
+    def purl(self):
+        return PackageURL(
+                    type='opam',
+                    name=self.name
+                ).to_string()
+
+
+# Regex expressions to parse file lines
+parse_file_line = re.compile(
+    r'(?P<key>^(.+?))'
+    r'\:\s*'
+    r'(?P<value>(.*))'
+).match
+
+parse_dep = re.compile(
+    r'^\s*\"'
+    r'(?P<name>[A-z0-9\-]*)'
+    r'\"\s*'
+    r'(?P<version>(.*))'
+).match
+
+"""
+Example:
+>>> p = parse_file_line('authors: "BAP Team"')
+>>> assert p.group('key') == ('authors')
+>>> assert p.group('value') == ('"BAP Team"')
+
+>>> p = parse_dep('"bap-std" {= "1.0.0"}')
+>>> assert p.group('name') == ('bap-std')
+>>> assert p.group('version') == ('{= "1.0.0"}')
+"""
+
+def parse_opam(location):
+    """
+    Return a dictionary containing all the opam data.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+
+    opam_data = {}
+
+    for i, line in enumerate(lines):
+        parsed_line = parse_file_line(line)
+        if parsed_line:
+            key = parsed_line.group('key').strip()
+            value = parsed_line.group('value').strip()
+            if 'maintainer' == key:
+                stripped_val = value.strip('["] ')
+                stripped_val = stripped_val.split('" "')
+                opam_data[key] = stripped_val
+                continue
```

Comment:
is continue ever needed?

Reply:
Continue was needed because we were splitting value in if statement and that value can not be used another function which is used later. So now I have reordered them and continue is no longer needed.